lib/core/CStatistics.cc

```diff
@@ -67,23 +67,31 @@ CStat& CStatistics::stat(int index) {
     return ms_Instance.m_Stats[index];
 }
 
+void CStatistics::cacheStats() {
+    ms_Instance.m_Cache = std::make_unique<CStatsCache>();
+    ms_Instance.m_Cache->populate(ms_Instance.m_Stats);
```

Comment:
I left a comment elsewhere that the work of `populate()` should be done in the constructor.  That has a spin-off benefit here too, because then if this method is called simultaneously from two threads it probably won't cause a core dump - whichever assigns `m_Cache` last wins, but at least you get a single consistent cache.

Reply:
If as above this issue goes away, i.e. `ms_Instance.m_Cache = std::make_unique<TStatsArray>(ms_Instance.m_Stats);`